bin/rcup.in

```diff
@@ -148,20 +155,30 @@ handle_file() {
 
   if [ "$generate" -ne 1 -a -e "$dest" ]; then
     if is_identical "$src" "$dest"; then
-      $VERBOSE "identical $dest"
+      if [ "x$sigil" != "xX" ] && ! is_linked "$src" "$dest"; then
+        $VERBOSE "replacing identical but incorrectly linked $dest"
+        replace_file "$src" "$dest" "$sigil"
+      else
+        $VERBOSE "identical $dest"
+      fi
     elif [ $REPLACE_ALL -eq 1 ]; then
       replace_file "$src" "$dest" "$sigil"
     else
-      $PROMPT "overwrite ${dest}? [ynaq]"
-      read overwrite
-      case "$overwrite" in
-        a) REPLACE_ALL=1
-           replace_file "$src" "$dest" "$sigil"
-           ;;
-        y) replace_file "$src" "$dest" "$sigil" ;;
-        q) exit 1 ;;
-        *) $VERBOSE "skipping $dest" ;;
-      esac
+      overwrite=""
+      while [ -z "$overwrite" -o "$overwrite" = "d" ]
+      do
+        $PROMPT "overwrite ${dest} (d for diff)? [yndaq]"
```

Comment:
I like the rest of this patch. How about this kind of UX instead:

```
overwrite .foo/bar? [yndaq?] ?
y - just this once
n - skip
d - show difference
a - yes and never ask again
q - quit
overwrite .foo/bar? [yndaq?] d
...
overwrite .foo/bar? [yndaq?] y
...
```

That is, add a `?` option that shows the help. This is more similar to other places where I've seen this (e.g. sysmerge(8) on OpenBSD, parts of git(1)).